### ISSUE TITLE
field access generalized identifiers highlight as seperate tokens

### ISSUE THREAD

**Issue by ninmonkey** [microsoft/vscode-powerquery#66]
## Summary

This is minor: field access without quoted-identifiers highlight as separate tokens.
ex: The textmate scope of `in` in `[Records in Range]` is a language keyword

![image](https://user-images.githubusercontent.com/3892031/100387862-b1676f80-2fee-11eb-8123-f4a5c3ebfc14.png)

I wasn't sure whether this is at the extension level or [microsoft/powerquery-parser](https://github.com/microsoft/powerquery-parser) I put it here because even though textmate's scope is a keyword -- the parser seems to format correctly. I tried using a strange `generalized-identifier` to see if I could break the formatting, but it did not. 

![record textmate scopes](https://user-images.githubusercontent.com/3892031/100387400-9e07d480-2fed-11eb-896c-ab669b8e1e0a.gif)

Power BI's editor has the same behaviour

![image](https://user-images.githubusercontent.com/3892031/100387477-ca235580-2fed-11eb-8cfc-7b6beb240df1.png)

## Test Query 

```powerquery
let
    Source = ...,
    #"Total Records1" = Table.AddColumn(
        Source, "Count 1",
        each Table.RowCount( [#"Records in Range"] ),
        type number
    ),
    #"Total Records2" = Table.AddColumn(
        Source, "Count 2",
        each Table.RowCount( [Records in Range] ),
        type number
    )

in
    #"Total Records2"
```

<details><summary>

## I think this is the grammar involved (Expand)
</summary>

 .

![image](https://user-images.githubusercontent.com/3892031/100388383-08217900-2ff0-11eb-9c07-496cb2ffc166.png)

</details>

**Comment by mattmasson** [comment]
Yeah, this is an issue with [our grammar file](https://github.com/microsoft/vscode-powerquery/blob/master/syntaxes/powerquery.tmLanguage.json). In theory we could resolve it by enabling semantic highlighting via the powerquery-parser, but it should be pretty straightforward to resolve.

FWIW, we do have a separate github project for the language grammar (that has fallen out of date), but it does have some built in tests. I've been meaning to get it running again, but if anyone else wants to pick that up, please feel free :) 

[powerquery-language](https://github.com/microsoft/powerquery-language)

/@JordanBoltonMN as FYI

**Comment by ninmonkey** [comment]
@mattmasson Which project is that?

**Comment by JordanBoltonMN** [comment]
> @mattmasson Which project is that?

https://github.com/microsoft/powerquery-language
